scylla/src/transport/load_balancing/mod.rs

```diff
@@ -53,10 +53,21 @@ pub trait ChildLoadBalancingPolicy: LoadBalancingPolicy {
     ) -> Box<dyn Iterator<Item = Arc<Node>> + Send + Sync>;
 }
 
-// Does safe modulo
-fn compute_rotation(index: usize, count: usize) -> usize {
-    if count != 0 {
-        index % count
+// Hashing round robin's index is a mitigation to problems that occur when a
+// `RoundRobin::apply_child_policy()` is called twice by a parent policy.
+fn round_robin_index_hash(index: usize) -> u64 {
+    let mut hasher = DefaultHasher::new();
+    hasher.write_usize(index);
+
+    hasher.finish()
+}
+
+// Does safe modulo and additionally hashes the index
+fn compute_rotation(round_robin_index: usize, sequence_length: usize) -> usize {
+    if sequence_length != 0 {
```

Comment:
Itty bitty nitpick: change this condition to `sequence_length <= 1`, with RF=1 we don't have to do this hashing at all and will always return 0.

Reply:
(oops, I made a mistake in the condition, hopefully it's correct now)